modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/IgniteResource.java

```diff
@@ -39,62 +39,65 @@
 
-    @Resources.BaseMessage("System view {0} is not modifiable")
+    @BaseMessage("System view {0} is not modifiable")
     ExInst<SqlValidatorException> systemViewIsNotModifiable(String systemViewName);
 
-    @Resources.BaseMessage("Ambiguous operator {0}. Dynamic parameter requires adding explicit type cast")
+    @BaseMessage("Ambiguous operator {0}. Dynamic parameter requires adding explicit type cast")
     Resources.ExInst<SqlValidatorException> ambiguousOperator1(String signature);
 
-    @Resources.BaseMessage("Ambiguous operator {0}. Dynamic parameter requires adding explicit type cast. Supported form(s): \n{1}")
+    @BaseMessage("Ambiguous operator {0}. Dynamic parameter requires adding explicit type cast. Supported form(s): \n{1}")
     Resources.ExInst<SqlValidatorException> ambiguousOperator2(String signature, String allowedSignatures);
 
-    @Resources.BaseMessage("Unable to determine type of a dynamic parameter. Dynamic parameter requires adding explicit type cast")
+    @BaseMessage("Unable to determine type of a dynamic parameter. Dynamic parameter requires adding explicit type cast")
     Resources.ExInst<SqlValidatorException> unableToResolveDynamicParameterType();
 
-    @Resources.BaseMessage("Incorrect type of a dynamic parameter. Expected <{0}> but got <{1}>")
+    @BaseMessage("Incorrect type of a dynamic parameter. Expected <{0}> but got <{1}>")
     Resources.ExInst<SqlValidatorException> incorrectDynamicParameterType(String expected, String actual);
 
-    @Resources.BaseMessage("Expression is not supported: {0}")
+    @BaseMessage("Expression is not supported: {0}")
     Resources.ExInst<SqlValidatorException> unsupportedExpression(String exprType);
 
-    @Resources.BaseMessage("CHAR datatype is not supported in table")
+    @BaseMessage("CHAR datatype is not supported in table")
     Resources.ExInst<SqlValidatorException> charDataTypeIsNotSupportedInTable();
 
     @BaseMessage("{0} datatype is not supported'")
     ExInst<SqlValidatorException> dataTypeIsNotSupported(String a0);
 
-    @Resources.BaseMessage("Length for type {0} must be at least 1")
+    @BaseMessage("Length for type {0} must be at least 1")
     Resources.ExInst<SqlValidatorException> invalidStringLength(String typeName);
 
-    @Resources.BaseMessage("Column N#{0} matched using NATURAL keyword or USING clause "
+    @BaseMessage("Column N#{0} matched using NATURAL keyword or USING clause "
             + "has incompatible types in this context: ''{1}'' to ''{2}''")
     Resources.ExInst<SqlValidatorException> naturalOrUsingColumnNotCompatible(int num, String type1, String type2);
 
     @BaseMessage("Cannot apply ''{0}'' to arguments of type {1}.")
     ExInst<SqlValidatorException> canNotApplyOp2Type(String a0, String a1);
 
+    @BaseMessage("A recursive query is not supported.")
+    ExInst<SqlValidatorException> recursiveQueryIsNotSupported();
```

Comment:
Let's use single approach for whole class.
Ether we should use pattern `Resources.ExInst ...` or ``ExInst ...``

Reply:
chose the shortest option, done